compiler/src/main/java/com/juul/pommel/compiler/utils/Utils.kt

```diff
@@ -1,4 +1,4 @@
-package com.juul.pommel.compiler
+package com.juul.pommel.compiler.utils
```

Comment:
This file is called `Utils` but is actual just some std lib extensions should be move the `extensions` package but what should the file be called? `StdlibExtensions`? 

If this gets move the the `extensions` package then maybe this package should be renamed to `constants`? Seeing how it is just a bunch of constant values and properties

Reply:
I personally am not a fan of `Utils`. It's so general it doesn't guide developers as to what is in it. I mean, ya, utilities, but for what? It's just so general.

Looking at the functions in here:

| Function | Comment(s) |
|-|-|
| `Any.cast` | Only used in one file, just move it to that file as `private`. |
| `castEach` | Only used in one file, just move it to that file as `private`. |
| `applyEach` | Since it's used in multiple places, maybe renamed this file to `Iteratable`, to represent extensions for `Iteratable`? |

Some other comments:

Having a general use casting functions that suppresses the warnings feels really gross to me. You're essentially saying anyone who uses this is going to hide the unsafe nature of it. Way better to not suppress warning, so you can suppress at usage site with a comment explaining why the unsafe operation is ok.

Why is `applyEach` not an extension function for `Iteratable`? it's named as if it were.

Reply:
@twyatt I agree that `Utils` is not all descriptive and uninformative. I say that JetBrains called the file where the scoping functions live `StandardKt` not sure if that is any better to be honest. 

As far as the casting functions are concerned is that I wanted something that looked a bit cleaner than the following 

```kotlin
     val constructors = element.enclosedElements
            .filter { it.kind == ElementKind.CONSTRUCTOR }
            .filter { it.hasAnnotation(INJECT_ANNOTATION) }
            .map {it as ExecutableElement}
```

When writing the annotation I realized there are situation where you just need to cast to get the information you want from an element. As we know all constructors are ExecutableElement and all Annotations are TypeElement they should be safe to do and if for some reason it is not then kapt will just crash. You definitely could do the just do the casting in place and write a comment about why we need to do this. That way no one can just abuse the general casting functions and it would be more easily caught during the PR review. I definitely see your point matter of preference I guess. 

And `applyEach` is not on `Iteratable` because I wanted to a scoping that would iterate over a collection and then return the object itself similar to `apply` For example 

rather than writing this 

```kotlin
MyClass.Builer()
.apply {
  properties.forEach { addProperty(it) }
}
```

I can just do this instead. 

```kotlin
MyClass.Builer()
.applyEach(properties) {
    addProperty(it)
}
```

That way I don't nested scoping functions because it sometimes be hard to read nesting scoping functions. Hopefully that makes sense